engine/base/hash-map.lisp

```diff
@@ -3,16 +3,20 @@
 (defclass hash ()
   (internal))
 
-(defun make-hash ()
+(defun make-hash (&optional pairs)
   (let ((h (make-instance 'hash)))
-    (setf (slot-value h 'internal)
-          (make-hash-table))
+    (setf (slot-value h 'internal) (make-hash-table))
+    (set-hash h pairs)
     h))
 
 (defun get-hash (hash key)
   (gethash key (slot-value hash 'internal)))
 
-(defmacro set-hash (hash (key value))
-  `(setf
-    (gethash ,key (slot-value ,hash 'internal))
-    ,value))
+(defun set-hash (hash &optional pairs)
+  (dolist (pair pairs hash)
+    (setf (gethash (first pair) (slot-value hash 'internal))
+          (second pair))))
+
+(defun equal-hash (hash-1 hash-2)
```

Comment:
I think that this is not the best name for equality predicate.
In CL, the convention is to place the type we're checking in the first place in the function name: `string=`, `tree-equal`, and so on.

Reply:
It relates to all functions in this package though...